runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/WindmillTimerInternals.java

```diff
@@ -64,10 +65,8 @@ class WindmillTimerInternals implements TimerInternals {
   // though technically in Windmill this is only enforced per ID and namespace
   // and TimeDomain. This TimerInternals is scoped to a step and key, shared
   // across namespaces.
-  private final Table<String, StateNamespace, TimerData> timers = HashBasedTable.create();
-
-  // Map from timer id to whether it is to be deleted or set
-  private final Table<String, StateNamespace, Boolean> timerStillPresent = HashBasedTable.create();
+  private final Map<Entry<String /*ID*/, StateNamespace>, Entry<TimerData, Boolean>> timerMap =
```

Comment:
Alternatively maybe we could use a TreeMap with a custom comparator?
Then we could avoid some of the wrapper allocations

If not is there a reason to use LinkedHashMap instead of just HashMap? Do we need the stable iteration for any reason?

Reply:
Changed to HashMap.